File: searchnsort/bubble_sort.cpp
//AlgoExpert - Bubble Sort - Search and Sort

#include<vector>
using namespace std;

vector<int> bubbleSort(vector<int> array){
	if (array.empty()){
		return {};
	}
	bool isSorted = false;
	int counter = 0;
	while(!isSorted){
		isSorted = true;
		for(int i = 0; i < array.size() - 1 - counter; i++){
			if(array[i] > array[i + 1]){
				swap(array[i], array[i + 1]);
				isSorted = false;
			}
		}
		counter++;
	}
	return array;
}

File: searchnsort/bubble_sort.py
'''
AlgoExpert - Bubble Sort - Search and Sort
'''

def bubbleSort(array):
	is_sorted = False
	counter = 0
	while not is_sorted:
		is_sorted = True
		for i in range(len(array) - 1 - counter):
			if array[i] > array[i + 1]:
				swap(i, i , array)
				is_sorted = False
		counter += 1
	return array

def swap(i, j, array):
	array[i], array[j] = array[j], array[i]
